### PR TITLE
adds social login button

### PR DESCRIPTION
**Pull request by nickfranciosi**
takes icon, text, action. 
Can be configured for any social network login

### CODE REVIEW

File: src/components/socialLoginButton/index.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { rgb } from "../../utils/color";
+import iconFromString from "../../utils/icon";
+
+const styles = {
+  base: {
+    width: "100%",
+    maxWidth: "295px",
+    borderRadius: "100px",
+    border: `1px solid ${color.detailHeaderSmall}`,
+    background: "transparent",
```
Comment (thomasthesecond) [inline src/components/socialLoginButton/index.jsx]: Can we use `backgroundColor` here?

File: src/components/socialLoginButton/index.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    paddingLeft: "24px",
+    fontSize: "16px",
+    fontWeight: typography.fontWeightMedium,
+    display: "flex",
+    justifyContent: "flex-start",
+    alignItems: "center",
+    cursor: "pointer",
+    transition: "color .5s ease",
+    ":hover": {
+      background: `rgba(${rgb(color.activeBackgroundColor)}, 0.4)`,
```
Comment (thomasthesecond) [inline src/components/socialLoginButton/index.jsx]: And here, along with active and focus styles?

File: src/components/socialLoginButton/index.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+const SocialLoginButton = ({ action, iconName, iconProps, text, style }) => (
+  <button style={[styles.base, style && style]} onClick={action}>
+    {iconFromString(iconName, Object.assign({}, iconSettings, iconProps))}
+    <span style={styles.text}>{text}</span>
+  </button>
+);
+
+SocialLoginButton.propTypes = {
+  text: PropTypes.string.isRequired,
+  iconName: PropTypes.string.isRequired,
```
Comment (thomasthesecond) [inline src/components/socialLoginButton/index.jsx]: Can we `import Icon from "../icon";` at the top and then change this line to
`iconName: PropTypes.oneOf(Object.keys(Icon)).isRequired,`?

File: src/components/socialLoginButton/index.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  action: PropTypes.func,
```
Comment (thomasthesecond) [inline src/components/socialLoginButton/index.jsx]: Can this be named `onClick`?

File: src/components/socialLoginButton/index.jsx (hunk 2 above)
Comment (thomasthesecond) [inline src/components/socialLoginButton/index.jsx]: I've been adding these to focus styles to buttons and links:
```
outline: "1px lightgray dotted",
outlineOffset: "2px",
```

**Comment by nickfranciosi** [comment]
@thomasthesecond should be good now

**Pull request merged**